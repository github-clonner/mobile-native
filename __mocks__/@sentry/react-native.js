export const captureMessage = jest.fn();
export default {};
