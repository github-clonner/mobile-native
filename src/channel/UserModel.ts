import { observable, action, runInAction } from 'mobx';

import { MINDS_CDN_URI, GOOGLE_PLAY_STORE } from '../config/Config';
import api from '../common/services/api.service';
import BaseModel from '../common/BaseModel';
import ChannelService from './ChannelService';
import sessionService from '../common/services/session.service';
import apiService from '../common/services/api.service';
import logService from '../common/services/log.service';

//@ts-nocheck
export const USER_MODE_OPEN = 0;
export const USER_MODE_MODERATED = 1;
export const USER_MODE_CLOSED = 2;

/**
 * User model
 */
export default class UserModel extends BaseModel {
  merchant;
  /**
   * Eth wallet
   */
  eth_wallet: string = '';
  wire_rewards;
  sums;
  btc_address?: string;
  icontime!: string;
  username!: string;
  name!: string;
  admin = false;
  plus: boolean = false;
  verified: boolean = false;
  founder: boolean = false;
  rewards: boolean = false;
  last_accepted_tos: number = 0;
  carousels?: Array<any>;

  /**
   * @var {boolean}
   */
  @observable blocked;

  /**
   * @var {number}
   */
  @observable subscribers_count;

  /**
   * @var {number}
   */
  @observable impressions;

  /**
   * @var {boolean}
   */
  @observable subscribed;

  /**
   * @var {boolean}
   */
  @observable mature_visibility = false;

  /**
   * @var {boolean}
   */
  @observable pending_subscribe = false;

  /**
   * @var {numeric}
   */
  @observable mode = 0;

  /**
   * @var {boolean}
   */
  @observable email_confirmed = false;

  /**
   * Confirm email
   * @param {Object} params
   */
  confirmEmail = async (params) => {
    // call any api endpoint with the param
    try {
      await apiService.get('api/v2/entities/', { urn: this.urn, ...params });
      this.setEmailConfirmed(true);
      return true;
    } catch (error) {
      return false;
    }
  };

  /**
   * Get the user icon time
   */
  getOwnerIcontime(): string {
    if (sessionService.getUser().guid === this.guid) {
      return sessionService.getUser().icontime;
    } else {
      return this.icontime;
    }
  }

  @action
  toggleMatureVisibility() {
    if (GOOGLE_PLAY_STORE) return;
    this.mature_visibility = !this.mature_visibility;
  }

  @action
  async toggleSubscription() {
    const value = !this.subscribed;
    this.subscribed = value;
    try {
      const metadata = this.getClientMetadata();
      await ChannelService.toggleSubscription(this.guid, value, metadata);
      UserModel.events.emit('toggleSubscription', this);
    } catch (err) {
      runInAction(() => {
        this.subscribed = !value;
      });
      throw err;
    }
  }

  @action
  async toggleBlock(value: boolean | null = null) {
    value = value === null ? !this.blocked : value;

    try {
      await ChannelService.toggleBlock(this.guid, value);
      this.blocked = value;
    } catch (err) {
      this.blocked = !value;
      logService.exception('[ChannelStore] toggleBlock', err);
    }
  }

  @action
  setMode(value) {
    this.mode = value;
  }

  @action
  setEmailConfirmed(value) {
    this.email_confirmed = value;
  }

  /**
   * Is admin
   */
  isAdmin() {
    return this.admin;
  }

  /**
   * current user is owner of the channel
   */
  isOwner = () => {
    return sessionService.getUser().guid === this.guid;
  };

  /**
   * Get banner source
   */
  getBannerSource() {
    if (this.carousels) {
      return {
        uri: this.carousels[0].src,
      };
    }
    return {
      uri: `${MINDS_CDN_URI}fs/v1/banners/${this.guid}/fat/${this.icontime}`,
      headers: api.buildHeaders(),
    };
  }

  /**
   * Get avatar source
   * @param {string} size
   */
  getAvatarSource(size = 'medium') {
    return {
      uri: `${MINDS_CDN_URI}icon/${
        this.guid
      }/${size}/${this.getOwnerIcontime()}`,
      headers: api.buildHeaders(),
    };
  }

  /**
   * Has banner
   * @returns {boolean}
   */
  hasBanner() {
    return !!this.carousels;
  }

  /**
   * Is closed
   * @returns {boolean}
   */
  isClosed() {
    return this.mode === USER_MODE_CLOSED;
  }

  /**
   * Is open
   * @returns {boolean}
   */
  isOpen() {
    return this.mode === USER_MODE_OPEN;
  }

  /**
   * Is moderated
   * @returns {boolean}
   */
  isModerated() {
    return this.mode === USER_MODE_MODERATED;
  }

  /**
   * Is subscribed
   * @returns {boolean}
   */
  isSubscribed() {
    return !!this.subscribed;
  }

  /**
   * Request subscribe
   */
  async subscribeRequest() {
    if (this.pending_subscribe || this.mode !== USER_MODE_CLOSED) {
      return;
    }
    try {
      this.pending_subscribe = true;
      await apiService.put(`api/v2/subscriptions/outgoing/${this.guid}`);
    } catch (err) {
      this.pending_subscribe = false;
      logService.exception(err);
    }
  }

  /**
   * Cancel subscribe request
   */
  async cancelSubscribeRequest() {
    if (!this.pending_subscribe || this.mode !== USER_MODE_CLOSED) {
      return;
    }
    try {
      this.pending_subscribe = false;
      await apiService.delete(`api/v2/subscriptions/outgoing/${this.guid}`);
    } catch (err) {
      this.pending_subscribe = true;
      logService.exception(err);
    }
  }
}
