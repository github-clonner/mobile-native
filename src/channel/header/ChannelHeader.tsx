//@ts-nocheck
import React, { Component } from 'react';

import { View, Text, TextInput, Image, TouchableHighlight } from 'react-native';

import { observer, inject } from 'mobx-react';

import Icon from 'react-native-vector-icons/Ionicons';
import FastImage from 'react-native-fast-image';
import * as Progress from 'react-native-progress';
import ReadMore from 'react-native-read-more-text';

import abbrev from '../../common/helpers/abbrev';
import ChannelActions from '../ChannelActions';
import Tags from '../../common/components/Tags';
import i18n from '../../common/services/i18n.service';
import session from '../../common/services/session.service';
import Touchable from '../../common/components/Touchable';
import ChannelBadges from '../badges/ChannelBadges';
import { CommonStyle } from '../../styles/Common';
import imagePicker from '../../common/services/image-picker.service';
import withPreventDoubleTap from '../../common/components/PreventDoubleTap';
import CompleteProfile from './CompleteProfile';
import featuresService from '../../common/services/features.service';
import ThemedStyles from '../../styles/ThemedStyles';
import remoteAction from '../../common/RemoteAction';

// prevent accidental double tap in touchables
const TouchableHighlightCustom = withPreventDoubleTap(TouchableHighlight);
const TouchableCustom = withPreventDoubleTap(Touchable);

/**
 * Channel Header
 */
@inject('user', 'onboarding')
@observer
class ChannelHeader extends Component {
  ActionSheetRef;
  loaded;

  state = {
    preview_avatar: null,
    preview_banner: null,
    briefdescription: '',
    name: '',
    saving: false,
    edit: false,
  };

  uploads = {
    avatar: null,
    banner: null,
  };

  /**
   * Get Channel Banner
   */
  getBannerFromChannel() {
    if (this.state.preview_banner) {
      return { uri: this.state.preview_banner };
    }

    return this.props.store.channel.getBannerSource();
  }

  /**
   * Get Channel Avatar
   */
  getAvatar() {
    if (this.state.preview_avatar) {
      return { uri: this.state.preview_avatar };
    }

    return this.props.store.channel.getAvatarSource('large');
  }

  _navToSubscribers() {
    if (this.props.navigation) {
      this.props.navigation.push('Subscribers', {
        guid: this.props.store.channel.guid,
      });
    }
  }

  componentDidMount() {
    const isOwner = session.guid === this.props.store.channel.guid;
    if (isOwner) {
      this.props.onboarding.getProgress();
    }
  }

  onEditAction = async () => {
    let editing = this.state.edit,
      payload = null;

    if (editing) {
      payload = {
        briefdescription: this.state.briefdescription,
        name: this.state.name,
        avatar: this.uploads.avatar,
        banner: this.uploads.banner,
        mode: this.props.store.channel.mode || 0,
      };

      this.setState({ saving: true });

      await remoteAction(async () => {
        let response;

        try {
          response = await this.props.store.save(payload);
        } finally {
          this.setState({ saving: false });
        }

        if (response === true) {
          this.props.user.load();
          this.setState({ saving: false, edit: false });
          this.uploads = {
            avatar: null,
            banner: null,
          };
        } else {
          this.setState({ saving: false });
        }
      });
    } else {
      this.setState({
        edit: true,
        preview_avatar: null,
        preview_banner: null,
        briefdescription: this.props.store.channel.briefdescription,
        name: this.props.store.channel.name,
      });
    }
  };

  onViewScheduledAction = async () => {
    this.props.store.feedStore.toggleScheduled();
  };

  subscribe() {
    this.props.store.channel.toggleSubscription();
  }

  changeBannerAction = async () => {
    imagePicker
      .show(i18n.t('channel.selectBanner'), 'photo')
      .then((response) => {
        if (response) {
          this.selectMedia('banner', response);
        }
      })
      .catch((err) => {
        alert(err);
      });
  };

  changeAvatarAction = async () => {
    if (!this.state.edit) return;
    imagePicker
      .show(i18n.t('channel.selectAvatar'), 'photo')
      .then((response) => {
        if (response) {
          this.selectMedia('avatar', response);
        }
      })
      .catch((err) => {
        alert(err);
      });
  };

  selectMedia(type, file) {
    this.setState({
      [`preview_${type}`]: file.uri,
    });

    this.uploads[type] = file;
  }

  setBriefdescription = (briefdescription) =>
    this.setState({ briefdescription });
  setName = (name) => this.setState({ name });

  /**
   * Truncated footer render for description
   */
  _renderTruncatedFooter = (handlePress) => {
    return (
      <Text
        style={[
          CommonStyle.fontM,
          CommonStyle.colorPrimary,
          CommonStyle.marginTop2x,
        ]}
        onPress={handlePress}>
        {i18n.t('readMore')}
      </Text>
    );
  };
  /**
   * Revealed footer render for description
   */
  _renderRevealedFooter = (handlePress) => {
    return (
      <Text
        style={[
          CommonStyle.fontM,
          CommonStyle.colorPrimary,
          CommonStyle.marginTop2x,
        ]}
        onPress={handlePress}>
        {i18n.t('showLess')}
      </Text>
    );
  };

  /**
   * Render Header
   */
  render() {
    const isOwner = session.guid === this.props.store.channel.guid;
    const channel = this.props.store.channel;
    const styles = this.props.styles;
    const avatar = this.getAvatar();
    const iurl = this.getBannerFromChannel();
    const isUploading = this.props.store.isUploading;
    const isEditable = this.state.edit && !isUploading;

    const ImageCmp = this.state.preview_banner ? Image : FastImage;
    const theme = ThemedStyles.style;

    return (
      <View>
        <TouchableCustom
          onPress={this.changeBannerAction}
          disabled={!isEditable}>
          <ImageCmp
            source={iurl}
            style={styles.banner}
            resizeMode={FastImage.resizeMode.cover}
          />

          {isEditable && (
            <View style={styles.tapOverlayView}>
              <Icon name="md-create" size={30} color="#fff" />
            </View>
          )}
          {isUploading && this.props.store.bannerProgress ? (
            <View style={styles.tapOverlayView}>
              <Progress.Pie
                progress={this.props.store.bannerProgress}
                size={36}
              />
            </View>
          ) : null}
        </TouchableCustom>

        <ChannelBadges
          channel={channel}
          style={{ position: 'absolute', right: 5, top: 160 }}
        />

        <View style={[styles.headertextcontainer, theme.backgroundSecondary]}>
          <View style={styles.countercontainer}>
            <TouchableHighlightCustom
              underlayColor="transparent"
              style={[styles.counter]}
              onPress={() => {
                this._navToSubscribers();
              }}>
              <View style={styles.counter} testID="SubscribersView">
                <Text style={[styles.countertitle, theme.colorPrimaryText]}>
                  {i18n.t('subscribers').toUpperCase()}
                </Text>
                <Text style={[styles.countervalue, theme.colorPrimaryText]}>
                  {abbrev(channel.subscribers_count, 0)}
                </Text>
              </View>
            </TouchableHighlightCustom>
            <View style={styles.counter} testID="ViewsView">
              <Text style={[styles.countertitle, theme.colorPrimaryText]}>
                {i18n.t('views').toUpperCase()}
              </Text>
              <Text style={[styles.countervalue, theme.colorPrimaryText]}>
                {abbrev(channel.impressions, 0)}
              </Text>
            </View>
          </View>
          <View style={styles.namecontainer}>
            <View style={styles.namecol}>
              {isEditable && (
                <TextInput
                  placeholder={channel.username}
                  underlineColorAndroid="transparent"
                  style={styles.nameTextInput}
                  value={this.state.name}
                  onChangeText={this.setName}
                  testID="ChannelNameTextInput"
                />
              )}
              {!isEditable && (
                <View
                  style={{ flexDirection: 'row', alignItems: 'center' }}
                  testID="ChannelNameView">
                  <Text
                    style={[styles.name, theme.colorPrimaryText]}
                    ellipsizeMode="tail"
                    numberOfLines={1}>
                    {channel.name}
                  </Text>
                </View>
              )}
              <Text style={styles.username}>@{channel.username}</Text>
            </View>
          </View>
          <View style={styles.buttonscol}>
            <ChannelActions
              navigation={this.props.navigation}
              store={this.props.store}
              onEditAction={this.onEditAction}
              onViewScheduledAction={this.onViewScheduledAction}
              editing={this.state.edit}
              saving={this.state.saving}
            />
          </View>
          {isEditable && (
            <View style={styles.briefdescriptionTextInputView}>
              <TextInput
                placeholder={i18n.t('channel.briefDescription')}
                multiline={true}
                underlineColorAndroid="transparent"
                style={styles.briefdescriptionTextInput}
                value={this.state.briefdescription}
                onChangeText={this.setBriefdescription}
              />
            </View>
          )}
          {!isEditable && (
            <View style={CommonStyle.paddingTop2x}>
              <ReadMore
                numberOfLines={3}
                renderTruncatedFooter={this._renderTruncatedFooter}
                renderRevealedFooter={this._renderRevealedFooter}>
                <Tags navigation={this.props.navigation}>
                  {channel.briefdescription}
                </Tags>
              </ReadMore>
            </View>
          )}
          {!isEditable && channel.city ? (
            <View
              style={[
                CommonStyle.paddingTop2x,
                CommonStyle.flexContainer,
                CommonStyle.rowJustifyStart,
              ]}>
              <Icon name="md-pin" size={24} style={styles.name} />
              <Text style={CommonStyle.marginLeft1x}>{channel.city}</Text>
            </View>
          ) : null}
        </View>

        <TouchableCustom
          onPress={this.changeAvatarAction}
          style={styles.avatar}
          disabled={!isEditable}>
          <Image source={avatar} style={styles.wrappedAvatar} />

          {isEditable && (
            <View
              style={[styles.tapOverlayView, styles.wrappedAvatarOverlayView]}>
              <Icon name="md-create" size={30} color="#fff" />
            </View>
          )}
          {isUploading && this.props.store.avatarProgress ? (
            <View
              style={[styles.tapOverlayView, styles.wrappedAvatarOverlayView]}>
              <Progress.Pie
                progress={this.props.store.avatarProgress}
                size={36}
              />
            </View>
          ) : null}
        </TouchableCustom>
        {/* {isOwner && (
          <CompleteProfile progress={this.props.onboarding.percentage} />
        )} */}
      </View>
    );
  }
}

export default ChannelHeader;
