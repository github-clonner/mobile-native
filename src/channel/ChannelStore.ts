//@ts-nocheck
import { observable, action } from 'mobx';
import { cloneDeep } from 'lodash';

import channelService from './ChannelService';
import wireService from '../wire/WireService';
import ChannelFeedStore from './ChannelFeedStore';
import UserModel from './UserModel';
import logService from '../common/services/log.service';
import channelsService from '../common/services/channels.service';

/**
 * Channel Store
 */
export default class ChannelStore {
  guid = null;
  feedStore = null;

  // only observable by reference because UserModel already have the needed observables
  @observable.ref channel = {};

  @observable rewards = {};
  @observable active = false;
  @observable loaded = false;
  @observable isUploading = false;
  @observable avatarProgress = 0;
  @observable bannerProgress = 0;

  constructor(guid) {
    this.guid = guid;
    this.feedStore = new ChannelFeedStore(guid);
  }

  @action
  clear() {
    this.channel = {};
    this.rewards = {};
    this.avatarProgress = 0;
    this.bannerProgress = 0;
    this.isUploading = false;
  }

  @action
  markInactive() {
    this.active = false;
  }

  @action
  setChannel(channel, loaded = false) {
    channel = UserModel.checkOrCreate(channel);
    this.channel = channel;
    this.feedStore.setChannel(channel);
    this.active = true;
    this.guid = channel.guid;

    if (loaded) {
      this.loaded = loaded;
    }
  }

  @action
  async load(defaultChannel) {
    if (defaultChannel) {
      defaultChannel = cloneDeep(defaultChannel);
    }
    const channel = await channelsService.get(this.guid, defaultChannel);

    if (channel) {
      this.loaded = true;
      this.setChannel(channel);
      return channel;
    }
    return false;
  }

  @action
  loadrewards(guid) {
    wireService
      .rewards(guid)
      .then(
        action((rewards) => {
          if (rewards) {
            // merge rewards
            rewards.merged = rewards.money.concat(rewards.points);
            this.rewards = rewards;
          } else {
            this.rewards = {};
          }
        }),
      )
      .catch((err) => {
        logService.exception('[ChannelStore] loadrewards', err);
      });
  }

  @action
  setBannerProgress(value) {
    this.bannerProgress = value;
  }

  @action
  setAvatarProgress(value) {
    this.avatarProgress = value;
  }

  @action
  setIsUploading(value) {
    this.isUploading = value;
  }

  @action
  async uploadAvatar(avatar) {
    this.setIsUploading(true);
    this.avatarProgress = 0;

    let avatarResult;

    try {
      avatarResult = await channelService.upload(
        this.guid,
        'avatar',
        {
          uri: avatar.uri,
          type: avatar.type,
          name: avatar.fileName || 'avatar.jpg',
        },
        (e) => {
          this.setAvatarProgress(e.loaded / e.total);
        },
      );

      this.setAvatarProgress(100);
    } catch (error) {
      this.setAvatarProgress(0);
      throw error;
    } finally {
      this.setIsUploading(false);
    }
  }

  @action
  async save({ avatar, banner, ...data }) {
    this.bannerProgress = 0;

    try {
      if (avatar) {
        await this.uploadAvatar(avatar);
      }

      if (banner) {
        this.isUploading = true;
        const bannerResult = await channelService.upload(
          this.guid,
          'banner',
          {
            uri: banner.uri,
            type: banner.type,
            name: banner.fileName || 'banner.jpg',
          },
          (e) => {
            this.setBannerProgress(e.loaded / e.total);
          },
        );

        this.setBannerProgress(100);

        if (bannerResult.error) return bannerResult.error;
      }

      const result = await channelService.save(data);
      const success = result && result.status === 'success';

      if (success) {
        this.channel.name = data.name;
        this.channel.briefdescription = data.briefdescription;
      }

      return success;
    } catch (e) {
      throw e;
    } finally {
      this.setIsUploading(false);
    }
  }

  @action
  reset() {
    this.guid = null;
    this.feedStore = null;
    // only observable by reference because UserModel already have the needed observables
    this.channel = {};
    this.rewards = {};
    this.loaded = false;
    this.active = false;
    this.isUploading = false;
  }
}
