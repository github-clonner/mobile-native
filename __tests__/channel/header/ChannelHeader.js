import 'react-native';
import React from 'react';
//import { Platform, CameraRoll, TouchableOpacity } from "react-native";
import ChannelHeader from '../../../src/channel/header/ChannelHeader';
import ChannelStore from '../../../src/channel/ChannelStore';
import userFaker from '../../../__mocks__/fake/channel/UserFactory';
import UserStore from '../../../src/auth/UserStore';
import UserModel from '../../../src/channel/UserModel';
import session from '../../../src/common/services/session.service';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import { getStores } from '../../../AppStores';

jest.mock('../../../src/auth/UserStore');
jest.mock('../../../src/channel/ChannelStore');
jest.mock('../../../src/common/services/boosted-content.service');

const appStores = {
  user: {
    me: {},
    load: jest.fn(),
    setUser: jest.fn(),
  },
};
getStores.mockReturnValue(appStores);

/**
 * Tests
 */
describe('channel header component owner', () => {
  let store, userStore, component;

  const navigation = {
    navigate: jest.fn(),
    state: {
      params: { guid: 1 },
    },
  };

  beforeEach(() => {
    store = new ChannelStore();
    userStore = new UserStore();
    userStore.me = UserModel.create(userFaker('1'));
    navigation.navigate.mockClear();
    appStores.user = userStore;
    session.guid = 1;

    store.channel = UserModel.create(userFaker('1'));
    store.loaded = true;

    component = renderer.create(
      <ChannelHeader.wrappedComponent
        store={store}
        user={userStore}
        navigation={navigation}
        styles={{}}
      />,
    );
  });

  it('should render correctly', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('should enable edit, save and disable', async () => {
    const instance = component.getInstance();

    // run onEdit
    await instance.onEditAction();

    expect(instance.state.edit).toBe(true);

    expect(component.root.findAllByType('TextInput').length).toBe(2);

    store.save.mockResolvedValue(true);

    // run again
    await instance.onEditAction();

    expect(instance.state.edit).toBe(false);

    expect(component.root.findAllByType('TextInput').length).toBe(0);
    expect(store.save).toBeCalled();
  });
});
