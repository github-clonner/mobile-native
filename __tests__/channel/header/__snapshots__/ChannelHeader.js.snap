// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`channel header component owner should render correctly 1`] = `
<View>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          undefined,
        ]
      }
    >
      <FastImageView
        resizeMode="cover"
        source={
          Object {
            "headers": Object {
              "App-Version": "3.8.0",
              "Cache-Control": "no-cache, no-store, must-revalidate",
              "Pragma": "no-cache",
            },
            "uri": "https://cdn.minds.com/fs/v1/banners/1/fat/1523515420",
          }
        }
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      />
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        },
        Object {
          "position": "absolute",
          "right": 5,
          "top": 160,
        },
      ]
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 25,
          },
          Object {
            "color": "#fff",
            "marginLeft": 5,
            "paddingBottom": 5,
            "shadowOpacity": 0.35,
            "textShadowColor": "#000000",
            "textShadowOffset": Object {
              "height": 1,
              "width": 1,
            },
            "textShadowRadius": 5,
          },
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 25,
          },
          Object {
            "color": "#fff",
            "marginLeft": 5,
            "paddingBottom": 5,
            "shadowOpacity": 0.35,
            "textShadowColor": "#000000",
            "textShadowOffset": Object {
              "height": 1,
              "width": 1,
            },
            "textShadowRadius": 5,
          },
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>
  <View
    style={
      Array [
        undefined,
        undefined,
      ]
    }
  >
    <View>
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            undefined,
          ]
        }
      >
        <View
          style={null}
          testID="SubscribersView"
        >
          <Text
            style={
              Array [
                undefined,
                undefined,
              ]
            }
          >
            SUBSCRIBERS
          </Text>
          <Text
            style={
              Array [
                undefined,
                undefined,
              ]
            }
          >
            50
          </Text>
        </View>
      </View>
      <View
        testID="ViewsView"
      >
        <Text
          style={
            Array [
              undefined,
              undefined,
            ]
          }
        >
          VIEWS
        </Text>
        <Text
          style={
            Array [
              undefined,
              undefined,
            ]
          }
        >
          100
        </Text>
      </View>
    </View>
    <View>
      <View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
          testID="ChannelNameView"
        >
          <Text
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              Array [
                undefined,
                undefined,
              ]
            }
          >
            Some User
          </Text>
        </View>
        <Text>
          @
          someusermane
        </Text>
      </View>
    </View>
    <View>
      <View
        style={
          Array [
            Object {
              "flexDirection": "row",
              "justifyContent": "flex-end",
            },
            Object {
              "marginTop": 10,
            },
          ]
        }
      >
        <View
          accessibilityLabel="Edit your channel settings"
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#5DBAC0",
              "borderRadius": 36,
              "flexDirection": "row",
              "justifyContent": "center",
              "marginLeft": 5,
              "opacity": 1,
            }
          }
          testID="EditButton"
        >
          <Text
            style={
              Array [
                Object {
                  "color": "white",
                },
                undefined,
              ]
            }
          >
             
            Edit
             
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "paddingTop": 10,
        }
      }
    >
      <View
        style={
          Object {
            "height": undefined,
          }
        }
      >
        <Text
          numberOfLines={0}
        >
          <Text
            selectable={true}
          />
        </Text>
      </View>
    </View>
  </View>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Image />
  </View>
</View>
`;
